sos/report/plugins/telegraf.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (C) 2024 Marcin Wilk <marcin.wilk@canonical.com>
+#
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, IndependentPlugin
+
+
+class Telegraf(Plugin, IndependentPlugin):
+
+    short_desc = 'Telegraf, the metric collecting tool, plugin'
+
+    plugin_name = "telegraf"
+    profiles = ('observability',)
+    services = ('telegraf',)
+    files = (
+        '/etc/telegraf/',
+        '/etc/default/telegraf',
+    )
+
+    def setup(self):
+        # Collect data from 'files' var
+        super().setup()
+
+        # collect logs in addition to the above
+        if self.get_option("all_logs"):
+            self.add_copy_spec([
+                "/var/log/telegraf/",
+            ])
+        else:
+            self.add_copy_spec([
+                "/var/log/telegraf/*.log",
+            ])
+
+    def postproc(self):
+        protect_keys = [
+            "password",
+            "token",
+            "pwd",
+            "PASSWORD",
+            "TOKEN",
+            "PWD",
```

Comment:
Since we apply [`IGNORECASE` regexp](https://github.com/sosreport/sos/blob/main/sos/archive.py#L581), the latest three items are redundant.

Reply:
Indeed. Fixed now.